File: MyWhiskyShelf.Core/Aggregates/Distillery.cs
using MyWhiskyShelf.Core.Models;

namespace MyWhiskyShelf.Core.Aggregates;

public sealed record Distillery
{
    public Guid Id { get; init; }
    public required string Name { get; init; }
    public required Guid CountryId { get; init; }
    public string? CountryName { get; init; }
    public required Guid? RegionId { get; init; }
    public string? RegionName { get; init; }
    public required int Founded { get; init; }
    public required string Owner { get; init; }
    public required string Type { get; init; }
    public required string Description { get; init; }
    public required string TastingNotes { get; init; }
    public required FlavourProfile FlavourProfile { get; init; } = new();
    public required bool Active { get; init; }
}

File: MyWhiskyShelf.Frontend/src/lib/mappers/distillery.ts
// @ts-ignore
import { DistilleryResponse } from "@/api/distilleries/contracts/types";
// @ts-ignore
import { Distillery } from "@/lib/domain/types";
// @ts-ignore
import { DistilleryCardProps } from "@/components/distilleries/DistilleryCard";

export function mapToDistillery(response: DistilleryResponse) : Distillery {
    return {
        id: response.id,
        name: response.name,
        countryId: response.countryId,
        countryName: response.countryName,
        regionId: response.regionId ?? undefined,
        regionName: response.regionName ?? undefined,
        founded: response.founded ?? undefined,
        owner: response.owner ?? undefined,
        type: response.type ?? undefined,
        description: response.description ?? undefined,
        tastingNotes: response.tastingNotes ?? undefined,
        flavourProfile: response.flavourProfile ?? undefined,
        active: response.active
    };
}

export function mapToDistilleryCardProps(domain: Distillery) : DistilleryCardProps {
    return {
        id: domain.id,
        name: domain.name,
        countryName: domain.countryName,
        regionName: domain.regionName ?? undefined,
        founded: domain.founded ?? undefined,
        owner: domain.owner ?? undefined,
        type: domain.type,
        description: domain.description ?? undefined,
        tastingNotes: domain.tastingNotes ?? undefined,
    }
}